frappe/public/js/frappe/ui/filters/filter.js

```diff
@@ -74,7 +73,11 @@ frappe.ui.Filter = class {
 	make() {
 		this.filter_edit_area = $(
 			frappe.render_template("edit_filter", {
-				conditions: this.conditions,
+				conditions: [
+					...this.conditions,
+					...this.nested_set_conditions,
+					...this.conditions_from_config,
+				],
```

Comment:
A bit more refactoring is possible if we may leave out the condition select options from the template rendering step. They are being completely rebuilt anyway.